tests/HttpServerTest.php

```diff
@@ -353,4 +353,194 @@ public function testTwoTypesOfEncodingForStreaming()
 
         $this->connection->emit('data', array($request));
     }
+
+    public function testNonNumericContentLengthResultsInError()
+    {
+        $callback = function(RequestInterface $request) {
+            $promise = new Promise(function ($resolve, $reject) use ($request) {
+                $request->getBody()->on('end', function() use ($resolve) {
+                    $resolve(new Response());
+                });
+            });
+                return $promise;
+        };
+
+
+        $request = "GET / HTTP/1.1\r\nHost: me.you\r\nContent-Length: bla\r\n\r\n";
+
+        $socket = new Socket($this->loop);
+        $server = new HttpServer($socket, $callback);
+
+        $socket->emit('connection', array($this->connection));
+
+        $this->connection->expects($this->once())->method('write')->with($this->equalTo("HTTP/1.1 400 Bad Request\r\n\r\n"));
+
+        $this->connection->emit('data', array($request));
+    }
+
+    public function testMultipleValuesInContentLengthResultsInError()
+    {
+        $callback = function(RequestInterface $request) {
+            $promise = new Promise(function ($resolve, $reject) use ($request) {
+                $request->getBody()->on('end', function() use ($resolve) {
+                    $resolve(new Response());
+                });
+            });
+                return $promise;
+        };
```

Comment:
Is this callback even used? It's my understanding that this won't be invoked at all?